### ISSUE TITLE
Ctrl+Enter has unpredictable side-effects

### ISSUE THREAD

**Issue by hhmmjjnn** [dbeaver/dbeaver#22678]
### Description

In the **SQL Editor**, hitting the shortcut **Ctrl+Enter** to trigger the `org.jkiss.dbeaver.ui.editors.sql.run.statement` command *(which is the default mapping)* may have one of this side-effects:

1. If no text is selected, the text, from the beggining of the line in which the cursor is, to the cursor, may be deleted;
2. if any text is selected, the text, from the beggining of the line in which the selection starts, to the end of the selection, may be deleted;
3. if any text is selected, the selected portion may be deleted;
4. or, sometimes, you just get lucky, and nothing crazy happens.

![image](https://github.com/dbeaver/dbeaver/assets/133042358/c6f59d25-4044-4598-a644-cefda7bbe7ea)
![image](https://github.com/dbeaver/dbeaver/assets/133042358/39096534-a0bb-4faf-a51f-432188125bd0)

It's a lottery. You simply can't predict what outcome you're getting.

Sometimes, you get a streak of sameness, but it eventually resets to pure randomness.

According to my gut, these are some factors that may, or may not influence this behaviour:

1. Whether the selected text contains valid SQL, or not;
2. whether the **Results** panel is showing an error, or not; and
3. whether the text contains a semicolon, or not.

Triggering the same command by other means, like the toolbar button...

![image](https://github.com/dbeaver/dbeaver/assets/133042358/c7daaf90-f57e-4d4d-aa93-4a40dc6e9bde)

...or the window menu item...

![image](https://github.com/dbeaver/dbeaver/assets/133042358/d60ee08a-608d-43a0-b4ba-8808b62f57cc)

...never triggers any of these glitches:

But my favourite workaround is always hitting the **Home** key right before **Ctrl+Enter**. This trick cancels any selection **(Cases 2, and 3)** and leaves nothing to be deleted to the left of the cursor. **(Case 1)** So success is guaranteed. **(Case 4)**

It's muscle memory for me, already, but not any less annoying.

### DBeaver Version

DBeaver Community Edition 23.3.0.202312031432

### Operating System

Windows 11 (21H2/10.0.22000.2652)

### Database and driver

`oracle.jdbc.OracleDriver`

### Steps to reproduce

*(...)*

### Additional context

*(...)*

**Comment by Destrolaric** [comment]
@hhmmjjnn
Can you please provide footage showing your issue? Thank you.

**Comment by hhmmjjnn** [comment]
> @hhmmjjnn Can you please provide footage showing your issue? Thank you.

Sure!

**Oucome 1.** If no text is selected, the text, from the beggining of the line in which the cursor is, to the cursor, may be deleted;

https://github.com/dbeaver/dbeaver/assets/133042358/a5f42496-339e-43c6-9ef5-b647da69226c

**Oucome 2.** if any text is selected, the text, from the beggining of the line in which the selection starts, to the end of the selection, may be deleted;

https://github.com/dbeaver/dbeaver/assets/133042358/e7b6b0ca-4128-4176-b8e4-9afa3af292e5

**Oucome 3.** if any text is selected, the selected portion may be deleted;

*(failed to reproduce)*

**Oucome 4.** or, sometimes, you just get lucky, and nothing crazy happens.

*(failed to reproduce)*

**Comment by E1izabeth** [comment]
@hhmmjjnn could you check if you have some shortcut binding set for the command "Cut to Beginning of Line"?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/1cb40a47-7475-4c4d-b2f5-63e51e109a11)

**Comment by hhmmjjnn** [comment]
@E1izabeth

It seems to be just like yours:

![image](https://github.com/dbeaver/dbeaver/assets/133042358/c3449237-d571-476b-ae0a-18d6d9befb21)

Please note that before shooting the videos I attached, I created a shining brand-new "CLEAN" workspace with purely default settings:

![image](https://github.com/dbeaver/dbeaver/assets/133042358/8719524b-1f00-498e-a7ae-fc133bc4d2de)

It's also worth noting that "Keys" has been terribly mistranslated to "Chaves" (keys of a keyring) in the **pt-BR** locale. Should have been "Teclas" (keys of a piano) instead.

**Comment by hhmmjjnn** [comment]
It should be noted that, for **Outcome 1**, the "beggining of the line" is actually the first character past indentation, which might as well be to the right of the cursor:

https://github.com/dbeaver/dbeaver/assets/133042358/7139f1b7-2182-4b65-87e1-f6994e9cc1fd

Then, I have just managed to reproduce yet a fifth outcome:

**Outcome 5.** If some text is selected, it gets replaced by newline, then the indentation is deleted:

https://github.com/dbeaver/dbeaver/assets/133042358/40276c24-4f0e-4a25-89d8-a581d9b43986

That last one is particulary wierd because it takes two `Ctrl+Z` strokes to undo:

https://github.com/dbeaver/dbeaver/assets/133042358/46134d1f-d677-4f23-8778-444ad7f83ab1

**Comment by Destrolaric** [comment]
@hhmmjjnn We are still not able to reproduce, which keyboard layout do you use in Windows?

**Comment by Destrolaric** [comment]
> [@hhmmjjnn](https://github.com/hhmmjjnn) We are still not able to reproduce, which keyboard layout do you use in Windows?

@hhmmjjnn, Hi. Can you provide information about the question asked above

**Comment by hhmmjjnn** [comment]
Sorry @Destrolaric, I don't log into GitHub very often...

My keyboard layout is `PTB2`

![image](https://github.com/dbeaver/dbeaver/assets/133042358/90a62498-bdea-462d-b3c4-4035f408f696)

**Comment by hhmmjjnn** [comment]
Some observations:

1. I switched to the International layout, but it doesn't seem to make a difference;
2. I connected to another machine via RDP, and the bug doesn't happen inside the MSTC window; (both ends are `PTB2`)
3. If I use the `Enter` key from the numeric side of the keyboard, the bug doesn't seem to happen.

I'm currently investigating whether any third-party app might have globally hooked my central `Enter` key. (But, that wouldn't explain why MSTC is immune)
